chaste_codegen/chaste_model.py

```diff
@@ -321,33 +320,19 @@ def _get_cytosolic_calcium_concentration_var(self):
             self._logger.info(warning)
             return cytosolic_calcium_concentration
 
-    def _get_modifiable_parameters_annotated(self):
-        """ Get the variables annotated in the model as modifiable parametery"""
-        return self._model.get_variables_by_rdf((self._PYCMLMETA, 'modifiable-parameter'), 'yes')
-
-    def _get_modifiable_parameters_exposed(self):
-        """ Get the variables in the model that have exposed annotation and are modifiable parameters
-            (irrespective of any modifiable_parameters tags)"""
-        return [q for q in self._model.variables()
-                if self._model.has_ontology_annotation(q, self._OXMETA)
-                and not self._model.get_ontology_terms_by_variable(q, self._OXMETA)[-1] == 'membrane_stimulus_current'
-                and q not in self._model.get_derived_quantities()
-                and q not in self._state_vars
-                and not q == self._time_variable]
-
     def _get_modifiable_parameters(self):
-        """ Get all modifiable parameters
-
-            Note: the result depends on self.expose_annotated_variables to determine whether or not to include
-            variables exposed with oxford metadata that are a modifiable parameter but are not annotated as such"""
-        if self.expose_annotated_variables:
-            # Combined and sorted in display name order
-            return \
-                sorted(set(self._get_modifiable_parameters_annotated() + self._get_modifiable_parameters_exposed()),
-                       key=lambda v: self._model.get_display_name(v, self._OXMETA))
-        else:
-            return sorted(self._get_modifiable_parameters_annotated(),
-                          key=lambda v: self._model.get_display_name(v, self._OXMETA))
+        """ Get all modifiable parameters, either annotated as such or with other annotation.
+
+        Stimulus currents are ignored and the result is sorted by display name"""
+        return sorted(set([q for q in list(self._model.variables()) +
```

Comment:
This is a long comprehension! Would something like the following be more readable?
```python
annotated_vars = set(
    [var for var in self._model.variables() if self._model.has_ontology_annotation(var, self._OXMETA)] +
    self._model.get_variables_by_rdf((self._PYCMLMETA, 'modifiable-parameter'), 'yes'))
current_vars = set(
    [var for var in annotated_vars
     if self._model.get_ontology_terms_by_variable(var, self._OXMETA)[-1].startswith('membrane_stimulus_current')])
parameters = annotated_vars - current_vars - set(self._model.get_derived_quantities() + self._state_vars + [self._time_variable])
return sorted(parameters, key=lambda v: self._model.get_display_name(v, self._OXMETA))
```